src/closure_decorator.rs

```diff
@@ -286,6 +258,17 @@ impl VisitMut for ClosureDecorator {
   }
 }
 
+fn prepend<T>(to: &mut Vec<T>, pre: T) {
+  let mut buf = Vec::with_capacity(to.len() + 1);
+  // TODO: Optimize (maybe unsafe)
+
+  buf.push(pre);
+  buf.append(to);
+  debug_assert!(to.is_empty());
+
+  *to = buf
+}
+
```

Comment:
I keep seeing `splice` as an option, not sure how much of a difference it would make, but seems to be the rust way to prepend.

```rs
to.splice(0..0, iter::once(pre));
```

Reply:
```ts
use std::iter;

pub fn prepend<T>(to: &mut Vec<T>, pre: T) {
  *to = to.splice(0..0, iter::once(pre)).collect();
}
```

Reply:
A LinkedList seems like it would be better :)